src/core/main.c

```diff
@@ -74,6 +74,10 @@ static void load_ini_setting(void) {
     }
     g_setting.autoscan.source = ini_getl("autoscan", "source", SETTING_SOURCE_LAST, SETTING_INI);
     g_setting.autoscan.last_source = ini_getl("autoscan", "last_source", SETTING_SOURCE_HDZERO, SETTING_INI);
+	
+	//embedded osd mode 4x3 default
+	ini_gets("osd", "embedded_mode", "2", str sizeof(str), SETTING_INI);
+	g_setting.osd.embedded_mode = atoi(str);
 
```

Comment:
Indentation should be spaces. The project has a `.clang-format` file so any new changes should follow that format.

Also, you have not defined an `osd` structure with `embedded_mode` as a field in `page_common.h`, so this code will not compile as it is.

Perhaps `embedded_mode` is not the right name, maybe `aspect_ratio` would be better?